### PR TITLE
openrct2-ui cast fixes

### PR DESCRIPTION
**Pull request by janisozaur**
Part of https://github.com/OpenRCT2/OpenRCT2/issues/11158, done with clang-tidy and manual fixups

### CODE REVIEW

**Comment by janisozaur** [comment]
To be squashed

File: src/openrct2-ui/windows/MapGen.cpp [hunk 1]
```diff
@@ -906,7 +906,7 @@ static void window_mapgen_simplex_mouseup(rct_window* w, rct_widgetindex widgetI
 
             mapgenSettings.simplex_low = _simplex_low;
             mapgenSettings.simplex_high = _simplex_high;
-            mapgenSettings.simplex_base_freq = (static_cast<float>(_simplex_base_freq) / 100.00f);
+            mapgenSettings.simplex_base_freq = (static_cast<float>(_simplex_base_freq)) / 100.00f;
```
Comment (janisozaur) [inline src/openrct2-ui/windows/MapGen.cpp:909]: :bug:

File: src/openrct2/rct1/S4Importer.cpp [hunk 2]
```diff
@@ -527,7 +527,7 @@ class S4Importer final : public IParkImporter
                             EntryList* entries = GetEntryList(objectType);
 
                             // Check if there are spare entries available
-                            size_t maxEntries = static_cast<size_t>(object_entry_group_counts[objectType]);
+                            size_t maxEntries = (size_t)object_entry_group_counts[objectType];
```
Comment (janisozaur) [inline src/openrct2/rct1/S4Importer.cpp:530]: aargh, non-ui files were not supposed be part of this PR

File: src/openrct2-ui/windows/MapGen.cpp (hunk 1 above)
Comment (tupaschoal) [reply src/openrct2-ui/windows/MapGen.cpp:909]: Nice!

File: src/openrct2-ui/windows/EditorObjectSelection.cpp [hunk 3]
```diff
@@ -909,7 +909,7 @@ static void window_editor_object_selection_invalidate(rct_window* w)
             for (int32_t i = 0; i < 6; i++)
             {
                 if (_filter_flags & (1 << (_numSourceGameItems + i)))
-                    w->pressed_widgets |= 1ULL << (WIDX_FILTER_RIDE_TAB_TRANSPORT + i);
+                    w->pressed_widgets |= static_cast<uint64_t>(1ULL << (WIDX_FILTER_RIDE_TAB_TRANSPORT + i));
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/EditorObjectSelection.cpp:912]: Do we want this reverted?

File: src/openrct2-ui/windows/Park.cpp [hunk 4]
```diff
@@ -1431,7 +1431,7 @@ static void window_park_stats_paint(rct_window* w, rct_drawpixelinfo* dpi)
     y += LIST_ROW_HEIGHT;
 
     // Draw number of rides / attractions
-    if (w->list_information_type != 0xFFFF)
+    if (w->list_information_type != static_cast<uint16_t>(-1))
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Park.cpp:1434]: It used to be like that, but @Gymnasiast suggested the `0xFFFF`, do we want to go back?

File: src/openrct2-ui/windows/RideList.cpp [hunk 5]
```diff
@@ -686,7 +686,7 @@ static void window_ride_list_scrollpaint(rct_window* w, rct_drawpixelinfo* dpi,
                 break;
             case INFORMATION_TYPE_RUNNING_COST:
                 formatSecondary = STR_RIDE_LIST_RUNNING_COST_UNKNOWN;
-                if (ride->upkeep_cost != (money16)(uint16_t)0xFFFF)
+                if (ride->upkeep_cost != static_cast<money16>(static_cast<uint16_t>(0xFFFF)))
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/RideList.cpp:689]: Does it need the double cast here?

Comment (tupaschoal) [reply src/openrct2-ui/windows/RideList.cpp:689]: [Seems like it doesn't](https://godbolt.org/z/JGcB2o)

Comment (janisozaur) [reply src/openrct2-ui/windows/RideList.cpp:689]: Please provide a suggestion comment, I can then merge it

Comment (Gymnasiast) [reply src/openrct2-ui/windows/RideList.cpp:689]:
```suggestion
                if (ride->upkeep_cost != static_cast<money16>(0xFFFF))
```

**Pull request merged**